docs/compilers/CSharp/Compiler Breaking Changes - DotNet 5.md

```diff
@@ -58,3 +58,11 @@
             o is sbyte or short or int or long;
     ```
     Because the `and` and `or` combinators can follow a type pattern, the compiler interprets them as part of the pattern combinator rather than an identifier for the declaration pattern. Consequently, it is an error to use `or` or `and` as pattern variable identifiers starting with C# 9.0.
+
+4. The fix for https://github.com/dotnet/roslyn/issues/44067 generates correct (different) code.
+   In certain cases the compiler used to generate code whose behavior was ambiguous according
+   to the CLR's specification. The compiler used to produce the warning CS1957 in those cases.
+   The compiler now generates correct unambiguous code rather than reporting CS1957. Because
+   it is possible that the runtime behavior of a program will change due to the change in our code
+   generation strategy, this could be a breaking change. If your program did not elicit the warning
```

Comment:
>this could be a breaking change [](start = 24, length = 31)

I guess, given the current CLR behavior, the code could switch from overriding the right method to overriding the wrong method at runtime.

Reply:
> I guess, given the current CLR behavior, the code could switch from overriding the right method to overriding the wrong method at runtime.

I don't think we've characterized the bug sufficiently well to conclude that can happen.

Reply:
> I don't think we've characterized the bug sufficiently well to conclude that can happen.

I think we can conclude that from the tests that you added/modified. Or at least try to do that. If behavior was order dependent without an explicit override and explicit override changes it and also order dependent, then it is possible that things just might become worse, until proven otherwise. If we cannot prove otherwise, we should assume worse for the purpose of determining impact of the behavior change.

---
In reply to: [442936553](https://github.com/dotnet/roslyn/pull/44025#discussion_r442936553) [](ancestors = 442936553)